tests/quic_p2p.rs

```diff
@@ -0,0 +1,40 @@
+use quic_p2p::{Config, Event, QuicP2p};
+use std::net::{IpAddr, Ipv4Addr};
+use std::sync::mpsc;
+use unwrap::unwrap;
+
+#[test]
+fn successfull_connection_stores_peer_in_bootstrap_cache() {
+    let (ev_tx, _ev_rx) = mpsc::channel();
+    let mut peer1 = QuicP2p::with_config(
+        ev_tx,
+        Config {
+            port: Some(0),
+            ip: Some(IpAddr::V4(Ipv4Addr::LOCALHOST)),
+            ..Default::default()
+        },
+    );
+    unwrap!(peer1.start_listening());
+    let peer1_conn_info = unwrap!(peer1.our_connection_info());
+
+    let (ev_tx, ev_rx) = mpsc::channel();
+    let mut peer2 = QuicP2p::with_config(
+        ev_tx,
+        Config {
+            port: Some(0),
+            ip: Some(IpAddr::V4(Ipv4Addr::LOCALHOST)),
+            ..Default::default()
+        },
+    );
+    unwrap!(peer2.start_listening());
+    peer2.connect_to(peer1_conn_info.clone());
+
+    for event in ev_rx.iter() {
+        if let Event::ConnectedTo { .. } = event {
```

Comment:
Good to assert that peers indeed connected to the expected other peer (so make use of the info in the event)